File: game_client/Assets/Hotfix~/Hotfix/Assets/Protocol/GS2GC/p201_TileMatchOp/GS2GC_201_004_RetTileMatchInit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace Hotfix.GS2GC.p201_TileMatchOp
{

public class GS2GC_201_004_RetTileMatchInit : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 数据
/// </summary>
private Hotfix.Common.TileMatchObj.TileMatch_Info info;


public GS2GC_201_004_RetTileMatchInit() {
	info = new Hotfix.Common.TileMatchObj.TileMatch_Info();
}

public GS2GC_201_004_RetTileMatchInit(
	Hotfix.Common.TileMatchObj.TileMatch_Info _info
) {	info = _info;
}

public byte getMainOrder() { return (byte)201; }

public byte getSubOrder() { return (byte)4; }

/// <summary>
/// 数据
/// </summary>
public Hotfix.Common.TileMatchObj.TileMatch_Info getInfo() { return info; }
/// <summary>
/// 数据
/// </summary>
public void setInfo(Hotfix.Common.TileMatchObj.TileMatch_Info _info) { info = _info; }


public int GetBufSize() {
	int _size = 0;
	_size += 4 + info.GetBufSize();

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 2;
	_size += 4 + info.GetBufSize();

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	int _infoCustLen = _buf.getInt();
	int _infoCurPos = _buf.getCurPos();
	info.ReadUnzipBuf(_buf, _infoCurPos + _infoCustLen);
	_buf.setPosition(_infoCurPos + _infoCustLen);

}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putInt(info.GetBufSize());
	info.PutUnzipBuf(_buf);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)201);
	_buf.put((byte)4);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)201);
	_recBuf.put((byte)4);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("info").Append(":").Append(info == null ? "null" : info.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ActivitiesV01/src/Hotfix/V01/GS2GC/p201_TileMatchOp/GS2GC_201_004_RetTileMatchInit.java
package Hotfix.V01.GS2GC.p201_TileMatchOp;

import java.nio.ByteBuffer;
public class GS2GC_201_004_RetTileMatchInit implements ALBasicProtocolPack._IALProtocolStructure {
/** 数据 */
private Hotfix.V01.Common.TileMatchObj.TileMatch_Info info;


public GS2GC_201_004_RetTileMatchInit() {
	info = new Hotfix.V01.Common.TileMatchObj.TileMatch_Info();
}

public GS2GC_201_004_RetTileMatchInit(
	 Hotfix.V01.Common.TileMatchObj.TileMatch_Info _info
) {	info = _info;
}

public final byte getMainOrder() { return (byte)201; }

public final byte getSubOrder() { return (byte)4; }

/** 数据 */
public Hotfix.V01.Common.TileMatchObj.TileMatch_Info getInfo() { return info; }
/** 数据 */
public void setInfo(Hotfix.V01.Common.TileMatchObj.TileMatch_Info _info) { info = _info; }


public final int GetBufSize() {
	int _size = 0;
	_size += 4 + info.GetBufSize();

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 2;
	_size += 4 + info.GetBufSize();

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	int _infoCustLen = _buf.getInt();
	int _infoCurPos = _buf.position();
	info.ReadUnzipBuf(_buf, _infoCurPos + _infoCustLen);
	_buf.position(_infoCurPos + _infoCustLen);

}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putInt(info.GetBufSize());
	info.PutUnzipBuf(_buf);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)201);
	_buf.put((byte)4);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)201);
	_recBuf.put((byte)4);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

